formats/properties/commonMain/src/kotlinx/serialization/properties/Properties.kt

```diff
@@ -89,6 +101,26 @@ public sealed class Properties(
             return structure(descriptor).also { copyTagsTo(it) }
         }
 
+        final override fun <T> decodeSerializableValue(deserializer: DeserializationStrategy<T>): T {
+            val type = map["type"]?.toString()
+            if (deserializer is AbstractPolymorphicSerializer<*>) {
+                var actualSerializer: DeserializationStrategy<out Any>? = null
+
+                if (type != null) {
```

Comment:
Actually, `type == null` is also a valid case if someone defined [polymorphic default serializer](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/polymorphism.md#default-polymorphic-type-handler-for-deserialization) — it may return serializer even if type wasn't specified.

Also, I do not see the reasons to check if `actualSerializer == null` and the call `findPolymorphicSerializer` again just to throw exception. 

These four lines can be simplified to `actualSerializer = deserializer.findPolymorphicSerializer(this, type)`. Am I right?

(double-checking in Json happens due to the fact that first time we look _optimistically_ hoping that type is the first key in object and we wouldn't have to read whole object as tree into memory just to find type token)

Reply:
Yeah, you're totally right. I did not know you could define your own polymorphic serializer, so it makes no sense checking if type is null

Thanks for the tips!